crates/uv-resolver/src/resolver/urls.rs

```diff
@@ -25,155 +33,132 @@ impl Urls {
+                // Registry requirement
+                continue;
+            };
+            // With an override `anyio==0.0.0` and a requirements.txt entry `./anyio`, we still use
+            // the URL. See `allow_recursive_url_local_path_override_constraint`
+            urls.remove(&requirement.name);
+            let previous = overrides.insert(requirement.name.clone(), url.clone());
+            if let Some(previous) = previous {
+                if !same_resource(&previous.parsed_url, &url.parsed_url, git) {
+                    return Err(ResolveError::ConflictingOverrideUrls(
+                        requirement.name.clone(),
+                        previous.verbatim.verbatim().to_string(),
+                        url.verbatim.verbatim().to_string(),
+                    ));
                 }
             }
         }
 
-        Ok(Self(urls))
+        Ok(Self {
+            regular: urls,
+            overrides,
+        })
     }
 
-    /// Return the [`VerbatimUrl`] associated with the given package name, if any.
-    pub(crate) fn get(&self, package: &PackageName) -> Option<&VerbatimParsedUrl> {
-        self.0.get(package)
+    /// Return the allowed [`VerbatimUrl`]s for given package from regular requirements and
+    /// constraints (but not overrides), if any.
+    ///
+    /// It's more than one more URL if they are in different forks (or conflict after forking).
+    pub(crate) fn get_regular(&self, package: &PackageName) -> Option<&[VerbatimParsedUrl]> {
+        self.regular.get(package).map(Vec::as_slice)
     }
 
-    /// Returns `true` if the [`ParsedUrl`] instances point to the same resource.
-    pub(crate) fn same_resource(a: &ParsedUrl, b: &ParsedUrl, git: &GitResolver) -> bool {
-        match (a, b) {
-            (ParsedUrl::Archive(a), ParsedUrl::Archive(b)) => {
-                a.subdirectory == b.subdirectory
-                    && CanonicalUrl::new(&a.url) == CanonicalUrl::new(&b.url)
-            }
-            (ParsedUrl::Git(a), ParsedUrl::Git(b)) => {
-                a.subdirectory == b.subdirectory && git.same_ref(&a.url, &b.url)
-            }
-            (ParsedUrl::Path(a), ParsedUrl::Path(b)) => {
-                a.install_path == b.install_path
-                    || is_same_file(&a.install_path, &b.install_path).unwrap_or(false)
-            }
-            (ParsedUrl::Directory(a), ParsedUrl::Directory(b)) => {
-                a.install_path == b.install_path
-                    || is_same_file(&a.install_path, &b.install_path).unwrap_or(false)
-            }
-            _ => false,
+    /// Return the [`VerbatimUrl`] override for the given package, if any.
+    pub(crate) fn get_override(&self, package: &PackageName) -> Option<&VerbatimParsedUrl> {
+        self.overrides.get(package)
+    }
+
+    /// Check if a URL is allowed (known), and if so, return its canonical form.
+    pub(crate) fn canonicalize_allowed_url<'a>(
```

Comment:
Why does this not have to check overrides?

Reply:
We try `urls.get_override` before calling this method

Reply:
This code also changed entirely